src/tickit/core/components/component.py

```diff
@@ -71,25 +75,38 @@ def __call__(self) -> Component:
 class BaseComponent(Component):
     """A base class for compnents, implementing state interface related methods."""
 
-    state_consumer: StateConsumer[Input]
-    state_producer: StateProducer[Union[Interrupt, Output]]
+    state_consumer: StateConsumer[Union[Input, StopComponent]]
+    state_producer: StateProducer[Union[Interrupt, Output, ComponentException]]
```

Comment:
Would also be nice to have an alias for `Union[Interrupt, Output, ComponentException]`